readme.md

```diff
@@ -15,834 +14,1113 @@
+		return this;
+	}
 
-    ```javascript
+	CKEDITOR.tools.extend( Ctor.prototype, {
+		getFoo: function() {
+			return this.foo;
+		};
 
-    // 4.2.1
-    // Type coercion and evaluation notes
+		setFoo: function( val ) {
+			return ( this.foo = val );
+		};
+	} );
 
-    // Use === only if you have to - for example when you're dealing
-    // with user input or you need to distinguish between null and undefined.
-    // You can safely use == in most cases, especially
-    // when function's input is documented.
-    // It will be caller's fault if he passes wrong arguments.
+	// Expose our constructor to the global object.
+	CKEDITOR.ctor = Ctor;
+} )();
+```
 
+## 2.6. Naming
 
-    // 4.2.2
-    // Booleans, Truthies & Falsies
+You are not a human code compiler/compressor, so don't try to be one.
 
-    // Booleans:
-    true, false
+The following code is an example of egregious naming:
 
-    // Truthy:
-    "foo", 1
+```javascript
+function q(s) {
+	return document.querySelectorAll(s);
+}
+var i,a=[],els=q("#foo");
+for(i=0;i<els.length;i++){a.push(els[i]);}
+```
 
-    // Falsy:
-    "", 0, null, undefined, NaN, void 0
+Here's the same piece of logic, but with kinder, more thoughtful naming (and a readable structure):
+```javascript
+
+function query( selector ) {
+  return document.querySelectorAll( selector );
+}
+
+var id = 0,
+	elements = [],
+	matches = query( '#foo' ),
+	length = matches.length;
+
+for ( ; id < length; id++ ) {
+	elements.push( matches[ id ] );
+}
+```
+
+### 2.6.1. Notation
+
+#### 2.6.1.1. Variables, functions, objects
+
+Use `camelCase` notation for naming:
+
+* variables
+
+```javascript
+var camelCaseVariableName = 'foo!';
+```
+
+* functions
+```javascript
+function camelCaseFunctionName() {}
+```
+
+* methods
+```javascript
+object.prototype.camelCaseMethodName = function() {};
+```
+
+* objects
+```javascript
+var camelCaseObjectName = {
+	foo: 'foo',
+	bar: function() {}
+};
+```
 
-    ```
+* instances
+```javascript
+var camelCaseInstance = new Object();
+```
 
+* public members
+```javascript
+CKEDITOR.sth.camelCase = sth;
+```
 
-5. <a name="practical">Practical Style</a>
+#### 2.6.1.2. Types
 
-    ```javascript
+Use `PascalCase` notation for naming constructor function:
 
-    // 5.1.1
-    // A Practical Module
+```javascript
+function PascalCase () {}
 
-    ( function() {
-        var Module = ( function() {
-            var data = "secret";
+PascalCase.prototype = {
+	foo: function() {},
+	bar: function() {}
+};
+```
 
-            return {
-                // This is some boolean property.
-                bool: true,
-                // Some string value.
-                string: 'a string',
-                // An array property.
-                array: [ 1, 2, 3, 4 ],
-                // An object property.
-                object: {
-                    lang: 'en-Us'
-                },
+#### 2.6.1.3. Constants
```

Comment:
Agree about convention, but we don't really use `const` variables, so I don't think it's necessary (especially with `var` statement).

Reply:
It's not about specific `const` language keyword but much older constants convention.